base/expr.jl

```diff
@@ -574,15 +574,16 @@ macro generated(f)
     if isa(f, Expr) && (f.head === :function || is_short_function_def(f))
         body = f.args[2]
         lno = body.args[1]
+        tmp = gensym("tmp")
```

Comment:
Just a little shorter:
```suggestion
        @gensym tmp
```

Reply:
Actually I used this first but got a compilation error. Maybe I just have to run a `make clean` though, I'll try again.

Reply:
A wait, this might just be an issue with `@gensym` not being defined yet during bootstrap. In that case just leave it like this. It doesn't save much anyways.

Reply:
Compilation still fails with the following error message:
```shell
error during bootstrap:
LoadError(at "compiler/compiler.jl" line 3: LoadError(at "expr.jl" line 544: LoadError(at "expr.jl" line 577: UndefVarError(var=:error))))
ijl_undefined_var_error at /home/david/sources/julia/src/rtutils.c:132
ijl_get_binding_or_error at /home/david/sources/julia/src/module.c:387
unknown function (ip: 0x7f426f913ed5)
unknown function (ip: 0x7f426f913e33)
unknown function (ip: 0x7f426f913ac4)
unknown function (ip: 0x7f426f9139e4)
unknown function (ip: 0x7f426f913237)
jl_invoke_julia_macro at /home/david/sources/julia/src/ast.c:1018
jl_expand_macros at /home/david/sources/julia/src/ast.c:1074
jl_expand_macros at /home/david/sources/julia/src/ast.c:1117
jl_expand_macros at /home/david/sources/julia/src/ast.c:1117
jl_expand_macros at /home/david/sources/julia/src/ast.c:1117
jl_expand_macros at /home/david/sources/julia/src/ast.c:1117
jl_expand_macros at /home/david/sources/julia/src/ast.c:1084
ijl_expand_with_loc_warn at /home/david/sources/julia/src/ast.c:1180
jl_parse_eval_all at /home/david/sources/julia/src/toplevel.c:1019
ijl_load_ at /home/david/sources/julia/src/toplevel.c:1069
unknown function (ip: 0x7f426f91083d)
unknown function (ip: 0x7f426f910784)
jl_apply at /home/david/sources/julia/src/julia.h:1777 [inlined]
do_call at /home/david/sources/julia/src/interpreter.c:126
eval_value at /home/david/sources/julia/src/interpreter.c:215
eval_stmt_value at /home/david/sources/julia/src/interpreter.c:166 [inlined]
eval_body at /home/david/sources/julia/src/interpreter.c:612
jl_interpret_toplevel_thunk at /home/david/sources/julia/src/interpreter.c:744
top-level scope at compiler/compiler.jl:46
jl_toplevel_eval_flex at /home/david/sources/julia/src/toplevel.c:888
jl_eval_module_expr at /home/david/sources/julia/src/toplevel.c:195 [inlined]
jl_toplevel_eval_flex at /home/david/sources/julia/src/toplevel.c:675
ijl_toplevel_eval_in at /home/david/sources/julia/src/toplevel.c:947
unknown function (ip: 0x7f426f90fc0d)
jl_apply at /home/david/sources/julia/src/julia.h:1777 [inlined]
do_call at /home/david/sources/julia/src/interpreter.c:126
eval_value at /home/david/sources/julia/src/interpreter.c:215
eval_stmt_value at /home/david/sources/julia/src/interpreter.c:166 [inlined]
eval_body at /home/david/sources/julia/src/interpreter.c:612
jl_interpret_toplevel_thunk at /home/david/sources/julia/src/interpreter.c:744
top-level scope at compiler/compiler.jl:3
jl_toplevel_eval_flex at /home/david/sources/julia/src/toplevel.c:888
jl_parse_eval_all at /home/david/sources/julia/src/toplevel.c:1022
ijl_load_ at /home/david/sources/julia/src/toplevel.c:1069
ijl_load at /home/david/sources/julia/src/toplevel.c:1082
exec_program at /home/david/sources/julia/src/jlapi.c:521
true_main at /home/david/sources/julia/src/jlapi.c:574
jl_repl_entrypoint at /home/david/sources/julia/src/jlapi.c:706
main at /home/david/sources/julia/cli/loader_exe.c:59
__libc_start_main at /usr/lib/libc.so.6 (unknown line)
_start at /home/david/sources/julia/usr/bin/julia (unknown line)

make[1]: *** [sysimage.mk:61: /home/david/sources/julia/usr/lib/julia/corecompiler.ji] Error 1
make: *** [Makefile:82: julia-sysimg-ji] Error 2
```
Any idea if and how this can be fixed?

Reply:
Ah just saw your message now :slightly_smiling_face: OK, I'll leave it.

Reply:
Yeah, this is an issue with bootstrapping. We define `Core.Compiler` and `Base` bit by bit, so if code gets executed earlier in the process, you don't have access to all higher-level functions and macros yet.